### DISCUSSION TITLE
Hi how can i download all the provisioning profiles using fastlane?

### DISCUSSION THREAD

**Discussion by viveksamaga** [fastlane/fastlane#21937] [General]
We are working on Provsioning profile download automation where we have to download all the provisioning profile including xcode to be downloaded using fastlane + spaceship api.
Apple developer portal: 2 team ID's

So far am able to get the profiles which are from 1 team id and of type only "App Store" which are valid.
Using below command 
`fastlane sigh download_all  --api_key_path <PATH-TO-API-KEY>/*****.json`
I see it internally calls _sigh/lib/sigh/download_all.rb_ script and i see all the options there as i need with other type of profiles.
But how can i pass them in the above command to download the profiles.
Any pointers?

**Comment by jennie258fitz** [comment]
To download provisioning profiles for multiple team IDs and different profile types using fastlane sigh download_all, you can specify additional parameters in your command. Here’s how you can do it:

Specify Team ID: Use the --team_id parameter to specify the team ID.
Specify Profile Type: Use the --development, --adhoc, or --enterprise parameters to specify the type of profile you want to download.
Here’s an example command that includes these parameters:

fastlane sigh download_all --api_key_path <PATH-TO-API-KEY>/*****.json --team_id <TEAM_ID> --development

If you need to download profiles for multiple team IDs, you can run the command separately for each team ID [Receiptify](https://www.receiptify.net/)

fastlane sigh download_all --api_key_path <PATH-TO-API-KEY>/*****.json --team_id <TEAM_ID_1> --development
fastlane sigh download_all --api_key_path <PATH-TO-API-KEY>/*****.json --team_id <TEAM_ID_2> --development

You can also include the --download_xcode_profiles option if you want to download Xcode managed provisioning profiles:

fastlane sigh download_all --api_key_path <PATH-TO-API-KEY>/*****.json --team_id <TEAM_ID> --development --download_xcode_profiles

**Reply by cjwhitsitt** [reply]
Is there a way to do this in a Fastfile? `fastlane sigh download_all` seems to be a cli-specific command and doesn't include the typical parameter formatting.
